src/scenes/DetaildView/dvReducer.ts

```diff
@@ -1,56 +1,97 @@
+import FetchMovieById from '../../services/rest/fetchMovieById';
+import FetchMovies, { getRequest4Genre } from '../../services/rest/fetchMovies';
+import FetchProps from '../../services/rest/fetchProps';
+import Movies, { Movie } from '../../services/rest/movie';
+import { DBG, GET_ID, GET_PROP } from '../../utils';
+import createReducer, { ICase, ISwitch } from '../../utils/createReducer';
+import RootActions from '../Root/rootActions';
+import SearchResultActions from '../SearchResults/srActions';
+import DetailedViewActions from './dvActions';
+import DetailedViewState from './dvState';
 
 const stateInit: DetailedViewState = {
-    data:[],
-    results:[],
+    data: [],
+    results: [],
 };
 
 const initDetailedView = ({ state, payload }: ICase<DetailedViewState>): DetailedViewState => ({
     ...state,
 });
 
-const matchId = (id:number) => (movie:Movie) => movie.id === id;
+const matchId = (id: number) => (movie: Movie) => movie.id === id;
 
-const movieClicked = ({ state, payload }: ICase<DetailedViewState>): DetailedViewState => {
-    const newState = {
+const urlFilmId = ({ state, payload }: ICase<DetailedViewState>): DetailedViewState => (
+    // state = {
+    //     ...state,
+    //     id: GET_ID(payload),
+    // },
+    // FetchMovieById(payload as FetchProps<Movie>),
+    // state
+{
+    ...state,
+    id: GET_ID(payload),
+});
+
+const movieClicked = ({ state, payload }: ICase<DetailedViewState>): DetailedViewState => (
+    state = {
         ...state,
+        id: GET_ID(payload),
         detailedPanel:
             state.data.find(matchId(GET_ID(payload))) ||
             state.results.find(matchId(GET_ID(payload))),
-    };
-    GET_ID(payload) && FetchMovies({
-        ... payload,
-        request: getRequest4Genre(newState),
-    } as FetchProps<Movies>);
-    return newState;
-}
+    },
+    Number(GET_ID(payload)) && FetchMovies({
+        ...payload,
+        request: getRequest4Genre(state),
+    } as FetchProps<Movies>),
+    state
+);
+
 const clickSearchSuccess = ({ state, payload }: ICase<DetailedViewState>): DetailedViewState => ({
-    ... state,
+    ...state,
     data: (payload as Movies).data,
 });
 
 const genreLoadSuccess = ({ state, payload }: ICase<DetailedViewState>): DetailedViewState => ({
-    ... state,
+    ...state,
     results: (payload as Movies).data,
 });
-const genreLoadFail = ({ state, payload }: ICase<DetailedViewState>): DetailedViewState => (
+const loadFail = ({ state, payload }: ICase<DetailedViewState>): DetailedViewState => (
     DBG(),
     state
 );
 
+const urlFetchMovie = ({ state, payload }: ICase<DetailedViewState>): DetailedViewState => (
+    FetchMovieById(payload as FetchProps<Movie>),
+    state
+);
+
+const urlFetchMovieSuccess = ({ state, payload }: ICase<DetailedViewState>): DetailedViewState => (
+    state = {...state,
+        detailedPanel: GET_PROP(payload, 'movie'),
+    },
+    FetchMovies({
+        ...GET_PROP(payload, 'genreFetchProps'),

```

Comment:
There is a good utility library which worth to try out https://lodash.com/docs/4.17.11#get